easybuild/framework/easyblock.py

```diff
@@ -2720,20 +2720,66 @@ def prepare_step(self, start_dir=True, load_tc_deps_modules=True):
         if start_dir:
             self.guess_start_dir()
 
+    def _run_command_stack(self, commands, caller):
+        """Execute stack of commands in a shell"""
+        self.log.debug(f"Stack of commands to be executed: {commands}")
+
+        # make sure we have a list of commands
+        if not isinstance(commands, (list, tuple)):
+            error_msg = f"Invalid value for '{caller}', should be list or tuple of strings: {commands}"
+            raise EasyBuildError(error_msg)
+
+        for cmd in commands:
+            if not isinstance(cmd, str):
+                raise EasyBuildError(f"Invalid element in '{caller}', not a string: {cmd}")
+            run_shell_cmd(cmd)
+
+    def run_step_main_action(self, action, step_name, pre_cmds=None, pre_opts="", post_opts=""):
+        """Construct main command of step and execute it"""
+        if pre_cmds:
+            self._run_command_stack(pre_cmds, f"pre_{step_name}_cmds")
+
+        step_cmd = f"{pre_opts} {action} {post_opts}"
+        return run_shell_cmd(step_cmd)
+
     def configure_step(self):
-        """Configure build  (abstract method)."""
-        raise NotImplementedError
+        """Configure build."""
+        if self.cfg['configure_cmd'] is not None:
+            res = self.run_step_main_action(
+                self.cfg['configure_cmd'],
+                "configure",
+                self.cfg['pre_configure_cmds'],
+                self.cfg['preconfigopts'],
+                self.cfg['configopts'],
```

Comment:
This looks kind of weird without using named options, since `pre_configure_cmd` is specified after `configure_cmd`, so at the very least this should become:

```suggestion
                'configure',
                pre_cmds=self.cfg['pre_configure_cmds'],
                pre_opts=self.cfg['preconfigopts'],
                post_opts=self.cfg['configopts'],
```

Even better would be to use only named options, so also use `action='self.cfg['configure_cmd']'` and `action='configure'` (and making those both `None` by default in `run_step_main_action` and mandatory with an `is None` check in there.

Also, maybe rename `run_step_main_action` to `run_core_step`, and only pass a string like `configure`, let `run_core_step` figure out the rest (to use `configure_cmd`, `preconfigopts`, etc.)

Reply:
good idea, it is indeed easy to construct the commands from just the step name. Done in https://github.com/easybuilders/easybuild-framework/pull/4531/commits/e05a9850eca3dac2c6a7f3e0b47e08f3da870657